src/dotnet/CarbonAware.Plugins.JsonReaderPlugin.Tests/JsonReaderPluginTests.cs

```diff
@@ -0,0 +1,139 @@
+        Assert.AreEqual(4, result.Count());
+    }
+
+    [Test]
+    public async Task TestDataByLocation_WhenMultipleLocationsProvided()
+    {
+        var mockPlugin = SetupMockPlugin();
+
+        Dictionary<string, object> props = new Dictionary<string, object>();
+        props[CarbonAwareConstants.Locations] = new List<string> { "eastus", "westus"};
+        var plugin = mockPlugin.Object;
+        var result = await plugin.GetEmissionsDataAsync(props);
+        List<string> locations = (List<string>)props[CarbonAwareConstants.Locations];
+        Assert.AreEqual(3, result.Count());
+
+        foreach (var r in result) {
+             Assert.IsTrue(locations.Contains(r.Location));
+        }
+    }
+
+    [Test]
+    public async Task TestDataByLocationAndTimePeriod()
+    {
+        var mockPlugin = SetupMockPlugin();
+        
+        Dictionary<string, object> props = new Dictionary<string, object>();
+        props[CarbonAwareConstants.Locations] = new List<string> { "eastus"};
+        props[CarbonAwareConstants.Start] = "2021-08-09";
+        props[CarbonAwareConstants.End] = "2021-12-09";
+
+        var plugin = mockPlugin.Object;
+        var result = await plugin.GetEmissionsDataAsync(props);
+        
+        Assert.AreEqual(1, result.Count());
+        Assert.AreEqual("eastus", result.First().Location);
+    }
+
+    [Test]
+    public async Task TestData_WhenNoEndTimeSpecified()
+    {
+        var mockPlugin = SetupMockPlugin();
+        
+        Dictionary<string, object> props = new Dictionary<string, object>();
+        props[CarbonAwareConstants.Locations] = new List<string> { "eastus"};
+        props[CarbonAwareConstants.Start] = "2021-12-09";
+
+        var plugin = mockPlugin.Object;
+        var result = await plugin.GetEmissionsDataAsync(props);
+        
+        Assert.AreEqual(1, result.Count());
+        Assert.AreEqual("eastus", result.First().Location);
+        Assert.AreEqual(DateTime.Parse("2021-09-01"), result.First().Time);
+    }
+
+    [Test]
+    public async Task TestData_WhenNoMatchedCriteria()
+    {
+        var mockPlugin = SetupMockPlugin();
+        
+        Dictionary<string, object> props = new Dictionary<string, object>();
+        props[CarbonAwareConstants.Locations] = new List<string> { "xyz"};
+        
+        var plugin = mockPlugin.Object;
+        var result = await plugin.GetEmissionsDataAsync(props);
+        
+        Assert.AreEqual(0, result.Count());
+    }
+
+    private Mock<CarbonAwareJsonReaderPlugin> SetupMockPlugin() {
+        var logger = Mock.Of<ILogger<CarbonAwareJsonReaderPlugin>>();
+        var mockPlugin = new Mock<CarbonAwareJsonReaderPlugin>(logger);
+        
+        mockPlugin.Protected()
+            .Setup<List<EmissionsData>>("GetSampleJson")
+            .Returns(GetTestEmissionData())
+            .Verifiable();
+
+        return mockPlugin;
+    }
+    private List<EmissionsData> GetTestEmissionData() {
+        return new List<EmissionsData>() {

```

Comment:
@gfmatthews I am not sure I understand. These tests specifically test the filter logic based on the input parameters passed in the dictionary. Hence we decided to mock the emissions data instead of actually reading the actual json file. 
Are you suggesting we also test the reading and parsing of the json file (which would be more of an integration test)?

Reply:
Reader note: we resolved this with an offline conversation. My biggest concern was just trying to figure out maintainability for some future scenario when a future person is reviewing / updating this code. This code relies on the fact that the data here matches another json file but that's not explicitly called out.